### PR TITLE
Fixes #16643: Lots and lots of unecessary compliance computation when compliance cache is invalidate because of serialization when no serialization is asked

### PR DESCRIPTION
**Pull request by ncharles**
https://issues.rudder.io/issues/16643

### CODE REVIEW

**Comment by ncharles** [comment]
Memory impact seems crazy. But it looks like going to the rule page for the first time is painfully slow as well

**Comment by ncharles** [comment]
actually, perfs looks ok, it's just the first instanciation

**Comment by ncharles** [comment]
PR updated with a new commit

**Comment by ncharles** [comment]
PR updated with a new commit

File: webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/services/reports/ExecutionBatch.scala [hunk 1]
```diff
@@ -924,6 +924,27 @@ object ExecutionBatch extends Loggable {
...
+          val currentDirectives = currentStatusReports.directives
+
+          //don't keep directive that were removed between the two configs
+          val toKeep = runDirectives.filterKeys(k => currentDirectives.keySet.contains(k))
+
+          //now override currentDirective with the one to keep in currentReport
+          val updatedDirectives = currentDirectives ++ toKeep
+          val newCompliance = runStatusReport.copy(directives = updatedDirectives)
+
+          (newCompliance :: c, n)
```
Comment (ncharles) [inline webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/services/reports/ExecutionBatch.scala]: i'm a bit unsure about this change: I prepend in a list, so it should be better.

**Comment by ncharles** [comment]
Commit modified

File: webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/services/reports/ExecutionBatch.scala (hunk 1 above)
Comment (fanf) [reply webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/services/reports/ExecutionBatch.scala]: you can't be better than prepending a list, so it seems safe

**Comment by ncharles** [comment]
OK, squash merging this PR

**Comment by ncharles** [comment]
OK, merging this PR

**Pull request merged**